### DISCUSSION TITLE
How do I Attach to the Last Session

### DISCUSSION THREAD

**Discussion by ShiromMakkad** [zellij-org/zellij#3689] [Q&A]
I'd like to create an alias to attach to the last session used on the current machine. I know you can use `zellij a` but if there's multiple sessions, it just prints all the sessions. I want it to connect to the top session in the list, but I don't want to type out its name. Is this possible?

**Comment by Lockszmith-GH** [comment]
What are these other sessions?

As in, what is your workflow, and what is reconnecting to the last session solving?

**Reply by ShiromMakkad** [reply]
The first thing I do in the morning is attach to the last session I was working on the day before. When that's a named session, it's not a big deal to remember what I was working on, but when it's an unnamed session, it's a little annoying. 

Also, I almost always reconnect to the same session the next morning. Named session or not, I'd like a shortcut for it.

**Reply by Lockszmith-GH** [reply]
I basically use a single session by default - I name it `$USER@$(hostname)`, this way my login scripts make sure I'm on the default session automatically. For other sessions, if I created any, I'll use the session picker to change them.

Would that help? (I mean until someone either mentions how to get the latest sessions, or someone implements it)

**Reply by ShiromMakkad** [reply]
That makes sense. I use multiple sessions frequently enough that I'd prefer this to be built in functionality, but it's only a couple keystrokes in the morning without it. I'm sure I could make a shell script for it if need be.

**Reply by Lockszmith-GH** [reply]
I've been thinking about this - and you might get away with a function/script wrapping zellij, which writes the session name into `/tmp/zellij-$UID-session.name` or something similar.

**Reply by ShiromMakkad** [reply]
Or just run `zellij a` and use `awk` to grab the top entry and running `zellij a <top_entry>`. 

EDIT: `zellij a` shows sessions in a random order. This won't work..

**Reply by ShiromMakkad** [reply]
Another annoying thing about `zellij a` is that it shows exited sessions, which means I have to scroll up to even see what sessions are open at that moment :/

**Comment by peterjc** [comment]
What does your login script look like for this? i.e. How do you handle the session not existing yet:

```console
$ zellij attach "$USER@$(hostname)"
No session with the name 'XXXXXXX@XXXXXXX' found!
```

e.g. Do you just manually do this once:

```console
$ zellij --session "$USER@$(hostname)"
...
```

Or is this done with an if-statement via something ``zellij list-sessions | grep "$USER@$(hostname)"`` perhaps?

----

I think I have the same use-case: I'm considering using zellij for a couple of different remote servers that I connect too, where the connection can be unstable (typically the VPN times out, or I relocated an connect to a different WiFi). I think I just want one consistent session on each remote machine, to use when ever I log into it.

**Reply by Lockszmith-GH** [reply]
One consistent session is easy enough.

```shell
if [[ -z "$ZELLIJ_SESSION_NAME" ]]; then
    zellij attach -c $USER@$(hostname)
fi
```

First, test if you're already in zellij on this session, then if not, attach to an existing session named `$USER@$(hostname)`, and because of `-c` if it doesn't exist, it will create it.

But it's different than 'the last session' OP is asking for, as they are using multiple sessions.
